New conceptual ai (#40997)
dotnet/docs: docs/ai/tutorials/tutorial-ai-vector-search.md

        }
        catch (Exception ex)
        {
            string message = $"OpenAIService.GetChatCompletionAsync(): {ex.Message}";
            Console.WriteLine(message);
            throw;
        }
    }
    ```

    The app also uses prompt engineering to ensure Open AI service limits and formats the response for supplied recipes.

    ```C#
    //System prompts to send with user prompts to instruct the model for chat session
    private readonly string _systemPromptRecipeAssistant = @"
        You are an intelligent assistant for Contoso Recipes. 
        You are designed to provide helpful answers to user questions about 
        recipes, cooking instructions provided in JSON format below.
  
        Instructions:
        - Only answer questions related to the recipe provided below,
        - Don't reference any recipe not provided below.
        - If you're unsure of an answer, say ""I don't know"" and recommend users search themselves.        
        - Your response  should be complete. 
        - List the Name of the Recipe at the start of your response followed by step by step cooking instructions.
        - Assume the user is not an expert in cooking.

>         - Format the content so that it can be printed to the Command Line console;

Reviewer: ```suggestion
        - Format the content so that it can be printed to the Command Line console.
```

>         - In case there are more than one recipes you find let the user pick the most appropiate recipe.";

Reviewer: ```suggestion
        - In case there is more than one recipe you find, let the user pick the most appropriate recipe.";
```
